### PR TITLE
[corlib] FileStream now uses SafeFileHandle.

### PR DESCRIPTION
**Pull request by esdrubal**
FileStream now stores a SafeFileHandle instead of a IntPtr handle.
Added ObjectDisposedException to more methods.
MonoIO ICalls used by FileStream now receives a SafeHandle instead of IntPtr.

### CODE REVIEW

**Comment by marek-safar** [comment]
Why not change MonoIO to use SafeFileHandle and let runtime do all the magic instead of manually wrapping it everywhere

File: mcs/class/corlib/System.IO/FileStream.cs [hunk 1]
```diff
@@ -159,11 +126,9 @@ public FileStream (SafeFileHandle handle, FileAccess access)
 		}
 		
 		[MonoLimitationAttribute("Need to use SafeFileHandle instead of underlying handle")]
```
Comment (marek-safar) [inline mcs/class/corlib/System.IO/FileStream.cs]: I guess this can be removed now

File: mono/metadata/icall-def.h [hunk 2]
```diff
@@ -332,33 +332,33 @@ ICALL(MONOIO_34, "DuplicateHandle", ves_icall_System_IO_MonoIO_DuplicateHandle)
 ICALL(MONOIO_37, "FindClose", ves_icall_System_IO_MonoIO_FindClose)
 ICALL(MONOIO_35, "FindFirst", ves_icall_System_IO_MonoIO_FindFirst)
 ICALL(MONOIO_36, "FindNext", ves_icall_System_IO_MonoIO_FindNext)
-ICALL(MONOIO_6, "Flush(intptr,System.IO.MonoIOError&)", ves_icall_System_IO_MonoIO_Flush)
+ICALL(MONOIO_6, "Flush(System.Runtime.InteropServices.SafeHandle,System.IO.MonoIOError&)", ves_icall_System_IO_MonoIO_Flush)
```
Comment (marek-safar) [inline mono/metadata/icall-def.h]: These need MONO_VERSION bump

**Comment by esdrubal** [comment]
Bumped MONO_CORLIB_VERSION to 115

**Comment by marek-safar** [comment]
You need to squash it for the bump to be correct

**Pull request merged**